Add support for 4k metal images

jlebon opened the pull request:
First, add a new `buildextend-metal4k` command to create 4k disk images.
Then, teach `kola` and `cosa run` to read these images.

To test:

    host$ cosa run -I metal4k
    ...
    vm$ sudo fdisk -l /dev/vda
    ...
    Sector size (logical/physical): 4096 bytes / 4096 bytes
    ...

One potentially controversial bit here is that this requires a newer
libguestfs which isn't in f31 yet, so we pull it from f32 for now.

Closes: coreos/fedora-coreos-tracker#385

jlebon commented:
Let's hold this until https://github.com/coreos/coreos-assembler/pull/1024 gets in, and I'll rebase on top of that.

jlebon commented:
Discussions about this in https://github.com/coreos/fedora-coreos-tracker/issues/385.

darkmuggle commented:
@jlebon when this is ready, let's merge this. I do not forsee #1024 merging quickly due to the larger question raised. This has user pain now and solving that is more important IMO.

bcg62 commented:
looks like #1024 was closed, is there anything else holding this up?

jlebon commented:
Got the artifact working, but need to complete wiring it up on the kola side to make it easier to test.

jlebon commented:
OK, dropped WIP on this and marked ready for review!

I updated the first comment with the commit message, but this is worth highlighting again:

> One potentially controversial bit here is that this requires a newer
> libguestfs which isn't in f31 yet, so we pull it from f32 for now.

I think this is probably fine since f32 is almost out. But given that we extensively use libguestfs, it's possible we'll hit regressions too.

cgwalters commented on src/cmd-run:
```diff
@@ -259,6 +259,14 @@ case "${DISK_CHANNEL}" in
 
+if [ "${IMAGE_TYPE}" == metal4k ]; then
+    kola_args+=("--qemu-native-4k")
+    # native 4k requires a UEFI bootloader
```
Hum.  Possibly we should drop the BIOS partition then in this case?

jlebon replied:
I was trying to keep it close to `metal`, but hmm yeah I think that makes sense.

cgwalters commented:
There's a lot of other stuff to enhance for this, including coreos-installer support and `testiso` too.  Let me keep holding a mutex on `testiso` stuff for now!

jlebon commented:
Updated!

cgwalters commented:
/approve
Notable change, so I'll defer to a second for lgtm.

jlebon commented:
Any more thoughts on this?

jlebon commented:
With the FCOS release that just went out, this would be a good time to get this in and let it bake and rollback the libguestfs bump if necessary.

cgwalters commented:
/lgtm
